build/common.sh

```diff
@@ -1061,537 +1031,8 @@ function kube::release::create_tarball() {
-    contents="$(curl -s "${public_link}")"
-  else
-    # If not public, validate using gsutil
-    kube::log::status "Validating uploaded version file at ${publish_file_dst}"
-    contents="$(gsutil cat "${publish_file_dst}")"
-  fi
-  if [[ "${contents}" == "${KUBE_GCS_PUBLISH_VERSION}" ]]; then
-    kube::log::status "Contents as expected: ${contents}"
-  else
-    kube::log::error "Expected contents of file to be ${KUBE_GCS_PUBLISH_VERSION}, but got ${contents}"
-    return 1
-  fi
-}
-
-# ---------------------------------------------------------------------------
-# Docker Release
-
-# Releases all docker images to a docker registry specified by KUBE_DOCKER_REGISTRY
-# using tag KUBE_DOCKER_IMAGE_TAG.
-#
-# Globals:
-#   KUBE_DOCKER_REGISTRY
-#   KUBE_DOCKER_IMAGE_TAG
-#   KUBE_SERVER_PLATFORMS
-# Returns:
-#   If new pushing docker images was successful.
-function kube::release::docker::release() {
-  local binaries=(
-    "kube-apiserver"
-    "kube-controller-manager"
-    "kube-scheduler"
-    "kube-proxy"
-    "hyperkube"
-  )
-
-  local docker_push_cmd=("${DOCKER[@]}")
-  if [[ "${KUBE_DOCKER_REGISTRY}" == "gcr.io/"* ]]; then
-    docker_push_cmd=("gcloud" "docker")
-  fi
-
-  if [[ "${KUBE_DOCKER_REGISTRY}" == "gcr.io/google_containers" ]]; then
-    # Activate credentials for the k8s.production.user@gmail.com
-    gcloud config set account k8s.production.user@gmail.com
-  fi
-
-  for arch in "${KUBE_SERVER_PLATFORMS[@]##*/}"; do
-    for binary in "${binaries[@]}"; do
-
-      # TODO(IBM): Enable hyperkube builds for ppc64le again
-      if [[ ${binary} != "hyperkube" || ${arch} != "ppc64le" ]]; then
-
-        local docker_target="${KUBE_DOCKER_REGISTRY}/${binary}-${arch}:${KUBE_DOCKER_IMAGE_TAG}"
-        kube::log::status "Pushing ${binary} to ${docker_target}"
-        "${docker_push_cmd[@]}" push "${docker_target}"
-
-        # If we have a amd64 docker image. Tag it without -amd64 also and push it for compatibility with earlier versions
-        if [[ ${arch} == "amd64" ]]; then
-          local legacy_docker_target="${KUBE_DOCKER_REGISTRY}/${binary}:${KUBE_DOCKER_IMAGE_TAG}"
-
-          "${DOCKER[@]}" tag -f "${docker_target}" "${legacy_docker_target}" 2>/dev/null
-
-          kube::log::status "Pushing ${binary} to ${legacy_docker_target}"
-          "${docker_push_cmd[@]}" push "${legacy_docker_target}"
-        fi
-      fi
-    done
-  done
-  if [[ "${KUBE_DOCKER_REGISTRY}" == "gcr.io/google_containers" ]]; then
-    # Activate default account
-    gcloud config set account ${USER}@google.com
-  fi
-}
-
-function kube::release::gcloud_account_is_active() {
-  local -r account="${1-}"
-  if [[ "$(gcloud config list --format='value(core.account)')" == "${account}" ]]; then
-    return 0
-  else
-    return 1
-  fi
-}
+###############################################################################
+# The ::release:: namespace functions have been moved to
```

Comment:
(not sure if we intend to rename the ::release:: functions here to something like ::package::?)